core/src/com/unciv/ui/screens/worldscreen/status/StatusButtons.kt

```diff
@@ -1,43 +1,48 @@
 package com.unciv.ui.screens.worldscreen.status
 
+import com.badlogic.gdx.scenes.scene2d.ui.Container
 import com.badlogic.gdx.scenes.scene2d.ui.HorizontalGroup
 import com.badlogic.gdx.utils.Disposable
 
 class StatusButtons(
-    nextTurnButton: NextTurnButton,
-    autoPlayStatusButton: AutoPlayStatusButton? = null,
-    multiplayerStatusButton: MultiplayerStatusButton? = null
+    val nextTurnButton: NextTurnButton
 ) : HorizontalGroup(), Disposable {
-    var autoPlayStatusButton: AutoPlayStatusButton? = autoPlayStatusButton
+    var autoPlayStatusButton: AutoPlayStatusButton? = null
         set(button) {
             autoPlayStatusButton?.remove()
             field = button
             if (button != null) {
                 addActorAt(0, button)
             }
         }
-    var multiplayerStatusButton: MultiplayerStatusButton? = multiplayerStatusButton
+    var multiplayerStatusButton: MultiplayerStatusButton? = null
         set(button) {
             multiplayerStatusButton?.remove()
             field = button
             if (button != null) {
                 addActorAt(0, button)
             }
         }
+    var unitWaitButton: UnitWaitButton? = null
+        set(button) {
+            // wait button is wrapped in container, remove that container
+            unitWaitButton?.parent?.remove()
+            field = button
+            if (button != null) {
+                // fix uneven spacing applied by HorizontalGroup.wrap()
+                val container = Container(button)
+                container.padBottom(nextTurnButton.height - button.height)
+                // insert next to next-turn-button
+                addActorAt(children.indexOf(nextTurnButton, true), container)
+            }
+        }
     
-
     init {
         space(10f)
         right()
         wrapReverse()
         wrapSpace(10f)
         rowRight()
-        if (autoPlayStatusButton != null) {
-            addActor(autoPlayStatusButton)
-        }
-        if (multiplayerStatusButton != null) {
-            addActor(multiplayerStatusButton)
-        }
```

Comment:
Did you remove these and not do a replacement? That's not good

Reply:
It's dead code